### ISSUE TITLE
Selector: escape sequences with spaces not allowed in unquoted identifiers

### ISSUE THREAD

**Issue by cosmo0** [jquery/jquery#4424]
### Description ###

Escaped selectors for attributes don't work using "find()", but they work using "$()".  
I would understand if they never worked, but I assumed "find()" and "$()" were doing the same thing...

As a workaround I'm quoting the value, but it's surprising.

Updated to jQuery 3.4.1 a few minutes ago.

### Link to test case ###

https://jsbin.com/cabimotovu/edit?js,console

````js
var selector = '[data-id=' + $.escapeSelector('0_some_$id') + ']';

console.log($(selector).length);

console.log($('html').find(selector).length);
````

Output:
```
 0
 "error"
 "Error: Syntax error, unrecognized expression: [data-id=\30 _some_\$id]
     at Function.Sizzle.error (https://code.jquery.com/jquery-git.js:1534:8)
     at Sizzle.tokenize (https://code.jquery.com/jquery-git.js:2190:11)
     at Function.Sizzle [as find] (https://code.jquery.com/jquery-git.js:807:15)
     at jQuery.fn.init.find (https://code.jquery.com/jquery-git.js:2871:11)
     at qawicop.js:5:48
     at https://static.jsbin.com/js/prod/runner-4.1.7.min.js:1:13924
     at https://static.jsbin.com/js/prod/runner-4.1.7.min.js:1:10866"
```

**Comment by mgol** [comment]
Thanks for the report.

You need to put the ID in quotes, escaping isn't always enough:
https://jsbin.com/murojut/1/edit?js,console
The first example works because it goes through `querySelectorAll` unchanged while in the second example selector has to be tokenized by Sizzle to split on commas & prepend temporary IDs to parent elements.

The only thing that puzzles me here is why the native example doesn't throw errors. We should figure out what's the reason of this difference in behavior & whether we should align.

**Comment by timmywil** [comment]
This is a valid bug in the tokenizer. Escaping the `0` results in `\30 ` with a space character, but that space is not a combinator but a valid part of the escape.

**Comment by mgol** [comment]
The Sizzle PR (https://github.com/jquery/sizzle/pull/450) has landed so I'm changing the milestone to `3.4.2`  & adding a proper label.

**Comment by mgol** [comment]
Sizzle update merged in #4641. The issue is now fixed both on `master` & on `3.x-stable`.
